src/Components/ProductTile.js

```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
-    setProducts(products.map((product) => product.id === products[index].id ? updatedProduct : product));
+
+  const username = ReactSession.get("username");
+  const [loggedin, setLoggedin] = useState(username !== '');
+  const [open, setOpen] = React.useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
   };
-  const downVote = () => {
-    const updatedProduct = { ...products[index] };
-    let currentVote = updatedProduct.upVoted ? 1 : (updatedProduct.downVoted ? -1 : 0);
-    updatedProduct.downVoted = !updatedProduct.downVoted;
-    updatedProduct.upVoted = false;
-    let newVote = updatedProduct.upVoted ? 1 : (updatedProduct.downVoted ? -1 : 0);
-    updatedProduct.votes = updatedProduct.votes - currentVote + newVote;
-    setProducts(products.map((product) => product.id === products[index].id ? updatedProduct : product));
+
+  const handleClose = () => {
+    setOpen(false);
   };
+
+  const increaseUpVote = () =>{
+    const form = new FormData();
+    console.log(products[index].uid);
+    form.append("uid", products[index].uid);
+    Service.post('addTotalVote', form)
+      .then(data => 
+        {
+          if(data.success)
+          {
+            const updatedProduct = { ...products[index] };
+            updatedProduct.votes = updatedProduct.votes + 1;
+            setProducts(products.map((product) => product.id === products[index].id ? updatedProduct : product));
+          }
+          console.log(data.code);
+          if (data.code > 200) {
+            console.log("Error")
+          } else {
+            console.log(data.success)
+          }
+        });
+  }
+  const handleUpVote = () =>{
+    const form = new FormData();
+    form.append("emailId", username);
+    form.append("productId", index);
+    Service.post('addVote', form)
+      .then(data => 
+        {
+          console.log(data.code);
+          if(data.success)
+          {
+            increaseUpVote();
+          }
+          if (data.code > 200) {
+            console.log("Error")
+          } else {
+            console.log(data.success)
+          }
+        });
+  }
+
+  const decreaseUpVote = () =>{
+    const form = new FormData();
+    form.append("uid", products[index].uid);
+    Service.post('subTotalVote', form)
+      .then(data => 
+        {
+          if(data.success)
+          {
+            const updatedProduct = { ...products[index] };
+            updatedProduct.votes = updatedProduct.votes - 1;
+            setProducts(products.map((product) => product.id === products[index].id ? updatedProduct : product));
+          }
+          console.log(data.code);
+          if (data.code > 200) {
+            console.log("Error")
+          } else {
+            console.log(data.success)
+          }
+        });
+  }
+  const handleDownVote = () =>{
+    const form = new FormData();
+    form.append("emailId", username);
+    form.append("productId", index);
+    Service.post('removeVote', form)
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
    Service.post("removeVote", form)
```